src/sql/src/plan/statement/ddl.rs

```diff
@@ -602,6 +605,76 @@ pub fn plan_create_source(
                     UnplannedSourceEnvelope::Upsert(UpsertStyle::Debezium { after_idx })
                 }
                 DbzMode::Plain => {
+                    // XXX(chae): should this be for upsert too?
+                    let tx_metadata = match with_option_objects.remove("tx_metadata") {
+                        Some(SqlOption::ObjectName {
+                            name: _,
+                            object_name: tx_metadata,
+                        }) => {
+                            // `with_option_objects` and `with_options` should correspond.  We want
+                            // to keep the `ObjectName` information but we also need to remove the
+                            // key from `with_options` so pass our validation below that all keys
+                            // are used.
+                            with_options.remove("tx_metadata").unwrap();
+
+                            let item = scx
+                                .catalog
+                                .resolve_item(&normalize::unresolved_object_name(tx_metadata)?)?;
+                            let full_name = scx.catalog.resolve_full_name(item.name());
+                            let mut desc_iter = item
+                                .desc(&full_name)
+                                .context("tx_metadata catalog item doesn't have a desc")?
+                                .iter();
+
+                            match desc_iter.next() {
```

Comment:
This validation is a bit different than the debezium one. In the debezium case we don't care about the column indexes, we just care that the columns have the correct name and type, regardless of their index. I'd vote for being consistent in what users are expected to offer